scheduler/src/cook/compute_cluster.clj

```diff
@@ -28,10 +28,14 @@
   (db-id [this]
     "Get a database entity-id for this compute cluster (used for putting it into a task structure).")
 
+  (initialize-cluster [this pool->fenzo pool->offers-chan]
+    "Initializes the cluster. Returns a channel that will be delivered on when the cluster loses leadership.")
+
+  (current-leader? [this]
+    "Returns true if this cook instance is currently the leader for the compute cluster")
+
   (get-mesos-driver-hack [this]
-    "Get the mesos driver. Hack; any funciton invoking this should be put within the compute-cluster implementation")
-  (set-mesos-driver-atom-hack! [this driver]
-    "Hack to overwrite the driver. Used until we fix the initialization order of compute-cluster"))
+    "Get the mesos driver. Hack; any funciton invoking this should be put within the compute-cluster implementation"))
```

Comment:
typo in `funciton`

Reply:
We should add a TODO here to clearly flag it as something that needs refactoring.

Reply:
Fixed typo, added todo.